x/stake/genesis.go

```diff
@@ -43,7 +43,7 @@ func InitGenesis(ctx sdk.Context, keeper Keeper, data types.GenesisState) (res [
 
 	for _, delegation := range data.Bonds {
 		keeper.SetDelegation(ctx, delegation)
-		keeper.PostDelegationCreated(ctx, delegation.DelegatorAddr, delegation.ValidatorAddr)
+		keeper.PreDelegationCreated(ctx, delegation.DelegatorAddr, delegation.ValidatorAddr)
```

Comment:
huh this is weird, I think we should call this before `SetDelegation`, this is inconsistent with how we call it in the keeper

(another reason to name the hooks properly!)

Reply:
Is that for this PR to fix, or another one:

Reply:
Done!